### DISCUSSION TITLE
referral program using @react-native-firebase/dynamic-links

### DISCUSSION THREAD

**Discussion by vineelk8** [invertase/react-native-firebase#4725] [Q&A]
I've followed all the steps to create a dynamic link [instruction](https://rnfirebase.io/dynamic-links/usage) using the `@react-native-firebase/dynamic-links` npm package, and everything is working fine as per the instructions provided. But after using the short link I'm unable to recognize who send the link and to whom I need to acknowledge the reward. Below are the details

Firebase console created links

Long Dynamic Link:

https://example.page.link/?link=https://www.example.com/&apn=com.example

Short Dynamic Link:

https://example.page.link/6SuK

Deep Link

https://www.example.com/

And I have created a short dynamic link from the mobile app side using the [instructions](https://firebase.google.com/docs/dynamic-links/rest) Created Dynamic Links with the REST API and the response of the rest API is as below

{"previewLink": "https://example.page.link/y1E4?d=1", "shortLink": "https://example.page.link/y1E4"}

And now I'm opening the app using the short link which got from the above response https://example.page.link/y1E4

I should get this https://example.page.link/y1E4 when I open the app but I'm getting the as below

{"minimumAppVersion": null, "url": "https://www.example.com/"}

My requirement is I'm working for a referral program when I will invite a friend to download the app. When someone downloads the app I should get a referral bonus. When the user downloads through the link I shared with a friend with this link https://example.page.link/y1E4 I can identify the referred user and acknowledge reward.

Please find a solution for me or a better solution if you have one, thanks in advance

**Comment by mikehardy** [comment]
Hi there, sorry this is not a problem with the module is it? I think this will be better asked on stackoverflow or from the upstream firebase docs, help here is 100% focused on the question "is the module working per the API docs or not?", and is not concerned so much with solution design
